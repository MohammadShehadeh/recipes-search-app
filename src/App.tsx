import { Hero } from './components/Hero';
import { Header } from './components/Header';
import { Search } from './components/Search';
import { Footer } from './components/Footer';
import { Route, Routes } from 'react-router-dom';
import { BrowserRouter } from 'react-router-dom';
import { Recipe } from './components/Recipe';

function App() {
  return (
    <>
      <Header />
      <main>
        <Hero />
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Search />} />
            <Route path="recipe/:id" element={<Recipe />} />
            <Route path="*" element={<Search />} />
          </Routes>
        </BrowserRouter>
      </main>
      <Footer />
    </>
  );
}

export default App;
