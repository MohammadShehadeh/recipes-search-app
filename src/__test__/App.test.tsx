import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import App from '../App';

describe('App', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Header correctly', () => {
    render(<App />);
    const headerElement = screen.getByText(/Search Recipes/i);
    expect(headerElement).toBeInTheDocument();
  });
});
