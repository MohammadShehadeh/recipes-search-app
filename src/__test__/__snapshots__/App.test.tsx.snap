// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render to match snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="wrapper main"
      >
        <a
          className="logo"
          href="/"
        >
          Search Recipes
        </a>
      </div>
    </div>
  </header>,
  <main>
    <section
      className="hero"
    >
      <div
        className="background"
        style={
          Object {
            "backgroundImage": "url('/food-hero.jpg')",
          }
        }
      >
        <div
          className="container"
        >
          <div
            className="wrapper main"
          >
            <span
              className="eyebrow contained"
            >
              <span
                className="label"
              >
                Healthy Lifestyle
              </span>
            </span>
            <h1
              className="headline"
            >
              Discover Culinary Excellence üçΩÔ∏è Your Gateway to Limitless Recipes!
            </h1>
          </div>
        </div>
      </div>
    </section>
    <svg
      className="separator"
      preserveAspectRatio="none"
      viewBox="0 0 1440 74"
    >
      <path
        d="M456.464 0.0433865C277.158 -1.70575 0 50.0141 0 50.0141V74H1440V50.0141C1440 50.0141 1320.4 31.1925 1243.09 27.0276C1099.33 19.2816 1019.08 53.1981 875.138 50.0141C710.527 46.3727 621.108 1.64949 456.464 0.0433865Z"
      />
    </svg>
    <section
      className="search"
    >
      <div
        className="container"
      >
        <div
          className="main"
        >
          <div
            className="grid"
          >
            <div
              className="col col-md-12 col-lg-8 col-md-order-2"
            >
              <span
                className="results"
              >
                0
                 Results
              </span>
              <h2
                className="title primary"
              >
                Search Results
              </h2>
              <div
                className="grid"
              />
            </div>
            <div
              className="col col-md-12 col-lg-4 col-md-order-1"
            >
              <form
                className="form"
                onSubmit={[Function]}
              >
                <fieldset
                  className="fieldset"
                >
                  <legend
                    className="legend"
                  >
                    Search For Recipes:
                  </legend>
                  <div
                    className="group"
                  >
                    <input
                      className="input"
                      id="search"
                      name="Search"
                      placeholder="Type recipe name..."
                      type="search"
                    />
                    <button
                      className="submit"
                      title="Search For Recipes"
                      type="submit"
                    >
                      <svg
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.5 14C7 14 5 12 5 9.5S7 5 9.5 5 14 7 14 9.5 12 14 9.5 14zm6 0h-.8l-.3-.3c1-1.1 1.6-2.6 1.6-4.2C16 5.9 13.1 3 9.5 3S3 5.9 3 9.5 5.9 16 9.5 16c1.6 0 3.1-.6 4.2-1.6l.3.3v.8l5 5 1.5-1.5-5-5z"
                        />
                      </svg>
                    </button>
                  </div>
                  <button
                    className="clear"
                    onClick={[Function]}
                    type="reset"
                  >
                    Clear
                  </button>
                </fieldset>
              </form>
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="container"
    >
      <div
        className="main"
      >
        Copyright ¬© 
        2023
         Mohammad Shehadeh. All rights reserved.
      </div>
    </div>
  </footer>,
]
`;
